src/pages/index.js

```diff
@@ -36,26 +133,107 @@ function handleImageClick(name, link) {
   imagePopup.open({ name, link });
 }
 
-// New PopupWithForm instances (userinfo integration change)
+// New PopupWithForm instances (userinfo integration change step 3)
 const editProfilePopup = new PopupWithForm("#profile-edit-modal", (data) => {
-  userInfo.setUserInfo({ name: data.title, job: data.description });
-  editProfilePopup.close(); //closes popup after editing profile
+  const submitButton = editProfilePopup._form.querySelector(".modal__button");
+  const originalButtonText = submitButton.textContent;
+  submitButton.textContent = "Saving...";
+
+  return api
+    .setUserInfo({
+      name: data.title,
+      about: data.description,
+    })
+    .then((updatedUserData) => {
+      userInfo.setUserInfo({
+        name: updatedUserData.name,
+        job: updatedUserData.about,
+      });
+      // return new Promise((resolve) => setTimeout(resolve, 1000)); // 1s delay
+    })
+    .catch((err) => {
+      console.error("Failed to update user info:", err);
+      throw err;
+    })
+    .finally(() => {
+      submitButton.textContent = originalButtonText;
+    });
 });
 //end
 
+//avatar edit popup step (8)
+const avatarEditPopup = new PopupWithForm("#avatar-edit-modal", (data) => {
+  const submitButton = avatarEditPopup._form.querySelector(".modal__button");
+  const originalButtonText = submitButton.textContent;
+  submitButton.textContent = "Saving...";
+
+  const avatarData = {
+    avatar: data["avatar-url"],
+  };
+
+  return api
+    .setUserAvatar(avatarData) // Return the promise chain
+    .then((updatedUserData) => {
+      console.log("Avatar update data:", updatedUserData); // Debug log
+      userInfo.setAvatar(updatedUserData.avatar);
+      // return new Promise((resolve) => setTimeout(resolve, 1000)); // 1000ms delay
+    })
+    .then(() => {
+      avatarEditPopup.close();
+      avatarEditFormValidator.disableButton();
+    })
+    .catch((err) => {
+      console.error("Failed to update avatar:", err);
+      throw err; // Re-throw the error so PopupWithForm can catch it
+    })
+    .finally(() => {
+      submitButton.textContent = originalButtonText;
+    });
+});
+avatarEditPopup.setEventListeners();
+// new addcard server integration (step 4)
 const addCardPopup = new PopupWithForm("#add-card-modal", (data) => {
-  const newCard = createCard({ name: data.title, link: data.description });
-  cardSection.addItem(newCard); // Add the new card using Section's method
-  addCardPopup.close(); //closes popup after adding card
-  addCardFormValidator.disableButton();
+  const submitButton = addCardPopup._form.querySelector(".modal__button");
```

Comment:
* You've already found the submit button inside `PopupWithForm`, so you can get it in `index.js`, but you should make this class field **public** by removing the `_` (underscrore)


So, your code will be much easier here


```js
  const submitButton = addCardPopup.submitButton

```